src/bot/commands/misc/convert.js

```diff
@@ -27,12 +27,50 @@ async function fetchLatestRates () {
  * @param {number} baseValue
  * @param {string} baseType Base unit
  * @param {string} targetType Target unit
- * @returns {strung} Success message
+ * @returns {string} Success message
  * @throws {Error} Generic error if conversion fails
  */
 function convertUnits (baseValue, baseType, targetType) {
 	const conversion = convert(baseValue).from(baseType).to(targetType);
-	return `${baseValue}${baseType} is equal to ${conversion.toFixed(2)}${targetType}`;
+	const fractionDigits = getFractionDigits(conversion);
+	return `${baseValue.toString()}${baseType} is equal to ${conversion.toFixed(fractionDigits)}${targetType}`;
+}
+
+/**
+ * Gets the count of zeroes after the decimal point but before a non-zero number
+ * @param {number} value
+ * @returns {number} Number of zeroes
+ */
+function getCountOfZeroesAfterDecimalPoint (value) {
+	let fractionalPart = value % 1;
+	let numberOfZeroes = -1;
+
+	while (fractionalPart < 1 && fractionalPart > 0) {
+		fractionalPart *= 10;
+		numberOfZeroes++;
+	}
+	numberOfZeroes = Math.max(0, numberOfZeroes);
+
+	return numberOfZeroes;
```

Comment:
I'd probably simplify this to just `return` the `Math.max` expression rather than doing another assignment:
```suggestion
	return Math.max(0, numberOfZeroes);
```